test/collection_serializer_test.rb

```diff
@@ -23,6 +23,21 @@ def build_named_collection(*resource)
         resource
       end
 
+      def build_named_collection_with_matching_serializer(*resource)
+        serialized_resource = Class.new(::Model)
+        serializer = Class.new(ActiveModel::Serializer) do
+          def json_key
+            'resources'
+          end
+        end
+
+        Object.const_set(:SerializedResource, serialized_resource)
+        Object.const_set(:SerializedResourceSerializer, serializer)
+
+        resource.define_singleton_method(:name) { 'SerializedResource' }
```

Comment:
Not sure how important to your use case, but note that in the test I changed 

```diff
- collection_serializer.new(model)
+ collection_serializer.new([model])
```

since the collection serializer should take in an array of models, or at least something enumerable.  Was that a bug in the test or in your implementation?